src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2201,14 +2240,17 @@ FindRouterWorkerList(List *shardIntervalList, bool shardsPresent,
  *
  * If round robin policy is set, the placement could be on any node in pg_dist_node.
  * Else, the local node is set for the placement.
+ *
+ * Queries can also involve local tables. In that case we always use the local
```

Comment:
What is the reason we create a local dummy placement if the query has a local table?

Reply:
I guess for something like
```
WITH r AS (
 SELECT random(), a, b
 FROM distributed_table) 
 INSERT INTO local_table SELECT a, b FROM r;
```

where the insert is rewritten as `INSERT INTO local_table select a, b FROM read_intermediate_results(...) ...`